boards/pimoroni-tufty2040/examples/tufty_blinky.rs

```diff
@@ -0,0 +1,166 @@
+//! # Tufty2040 Blinky Example
+//!
+//! Blinks the activity LED on a tufty2040 board, using an RP2040 Timer in Count-down mode.
+//!
+//! See the `Cargo.toml` file for Copyright and licence details.
+
+#![no_std]
+#![no_main]
+
+use pimoroni_tufty2040 as tufty;
+
+// The macro for our start-up function
+use tufty::entry;
+
+// GPIO traits
+use embedded_hal::digital::v2::{OutputPin, PinState};
+
+// Ensure we halt the program on panic (if we don't mention this crate it won't
+// be linked)
+use panic_halt as _;
+
+// A shorter alias for the Hardware Abstraction Layer, which provides
+// higher-level drivers.
+use tufty::hal;
+
+// A shorter alias for the Peripheral Access Crate, which provides low-level
+// register access
+use hal::pac;
+
+use hal::Timer;
+use hal::Clock;
+use hal::clocks::ClockSource;
+use hal::gpio::{PullNone, FunctionPio0};
+
+use tufty::DummyPin;
+
+// A few traits required for using the CountDown timer
+use embedded_hal::timer::CountDown;
+use embedded_graphics::draw_target::DrawTarget;
+use embedded_graphics::Drawable;
+use embedded_graphics::pixelcolor::{Rgb565, RgbColor};
+use embedded_graphics::geometry::Point;
+use embedded_graphics::primitives::{Circle, Primitive, PrimitiveStyleBuilder};
+use fugit::ExtU32;
+use st7789::ST7789;
+
+#[entry]
+fn main() -> ! {
+    // Grab our singleton objects
+    let mut pac = pac::Peripherals::take().unwrap();
+    let cp = pac::CorePeripherals::take().unwrap();
+
+    // Set up the watchdog driver - needed by the clock setup code
+    let mut watchdog = hal::Watchdog::new(pac.WATCHDOG);
+
+    // Configure the clocks
+    //
+    // The default is to generate a 125 MHz system clock
+    let clocks = hal::clocks::init_clocks_and_plls(
+        tufty::XOSC_CRYSTAL_FREQ,
+        pac.XOSC,
+        pac.CLOCKS,
+        pac.PLL_SYS,
+        pac.PLL_USB,
+        &mut pac.RESETS,
+        &mut watchdog,
+    )
+        .ok()
+        .unwrap();
+
+    // The single-cycle I/O block controls our GPIO pins
+    let sio = hal::Sio::new(pac.SIO);
+
+    // Set the pins up according to their function on this particular board
+    let pins = tufty::Pins::new(
+        pac.IO_BANK0,
+        pac.PADS_BANK0,
+        sio.gpio_bank0,
+        &mut pac.RESETS,
+    );
+
+    // Configure the timer peripheral to be a CountDown timer for our blinky delay
+    let timer = Timer::new(pac.TIMER, &mut pac.RESETS, &clocks);
+    let mut delay_timer = timer.count_down();
+
+    let mut delay = cortex_m::delay::Delay::new(cp.SYST, clocks.system_clock.get_freq().to_Hz());
+
+    // Set the LED to be an output
+    let mut led_pin = pins.led.into_push_pull_output();
+
+    pins.lcd_backlight.into_push_pull_output_in_state(PinState::High);
+    pins.lcd_rd.into_push_pull_output_in_state(PinState::High);
+
+    /*
+    Example of using GPIO data lines:
```

Comment:
Do you think anyone will use these pins as GPIO?
If so, it would be better to document this use-case elsewhere, not in an example.

I think everyone will use your PIO+DMA driver and never touch the display pins directly.

Reply:
Fair question - it was useful to me whilst getting everything working, but given just the built-in peripherals I don't know that anyone would choose to avoid the PIO unit & DMA channel.

Do you want me to consign the `GpioDataLines` implementation to history? (If I leave the `DisplayDataLines` trait, it's easy enough to implement out-of-tree.)